packages/stream_video_noise_cancellation/lib/stream_video_noise_cancellation_method_channel.dart

```diff
@@ -12,21 +13,41 @@ class MethodChannelStreamVideoNoiseCancellation
 
   @override
   Future<void> registerProcessor() async {
+    if (!CurrentPlatform.isAndroid && !CurrentPlatform.isIos) {
+      throw UnimplementedError(
+          'The current platform (${CurrentPlatform.name}) does not support audio processing.');
+    }
+
     await methodChannel.invokeMethod<void>('registerProcessor');
   }
 
   @override
   Future<bool?> isEnabled() {
+    if (!CurrentPlatform.isAndroid && !CurrentPlatform.isIos) {
+      throw UnimplementedError(
+          'The current platform (${CurrentPlatform.name}) does not support audio processing.');
+    }
+
     return methodChannel.invokeMethod<bool>('isEnabled');
   }
 
   @override
   Future<void> setEnabled(bool enabled) {
+    if (!CurrentPlatform.isAndroid && !CurrentPlatform.isIos) {
+      throw UnimplementedError(
+          'The current platform (${CurrentPlatform.name}) does not support audio processing.');
+    }
+
     return methodChannel.invokeMethod<void>('setEnabled', {'enabled': enabled});
   }
 
   @override
   Future<bool?> deviceSupportsAdvancedAudioProcessing() {
+    if (!CurrentPlatform.isAndroid && !CurrentPlatform.isIos) {
+      throw UnimplementedError(
+          'The current platform (${CurrentPlatform.name}) does not support audio processing.');
```

Comment:
I think here I would just return `false`. That way the other methods should also not be called any time. Maybe you could also return `false` in `isEnabled`. Because it's not just unsupported by the method channel, it's known to be not supported by the platform. The other methods I can agree on throwing an exception, but I think for the `deviceSupportsAdvancedAudioProcessing` it makes sense to just return false. Otherwise you call `deviceSupportsAdvancedAudioProcessing` and get `The current platform does not support audio processing.`, which is basically part of the question you are asking.

Reply:
Maybe we could keep it here as `UnimplementedError`, because it's the base method channel, but just return false in the `NoiseCancellationAudioProcessor` which is the public api.

> deviceSupportsAdvancedAudioProcessing != deviceSupportsNoiseCancellation 

I understand that not all iOS and android devices support AdvancedAudioProcessing, but on any other device (windows, macos, web) for sure it would be supported right? So in the `NoiseCancellationAudioProcessor` I would also return false for this method.